sql/logictest: expected FD determinant and dependants to be disjoint

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2421858&tab=buildLog) on [master@37653515b3b6a612b9ee71c4f3bcfb9511805edb](https://github.com/cockroachdb/cockroach/commits/37653515b3b6a612b9ee71c4f3bcfb9511805edb):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2232 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2220 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 1781 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 2315 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2141 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2046 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 1419 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1138 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2265 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2252 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 1799 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 2354 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2172 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2051 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 1437 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1140 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2317 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2293 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 1830 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 2517 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2236 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2192 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 1483 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1147 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2412 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2396 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 2046 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 2689 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2362 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2298 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 1623 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1156 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2536 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2521 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 2285 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 2880 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2468 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2504 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 1699 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 2640 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_10.test: 2508 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1158 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 3062 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 2550 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2632 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_26.test: 2599 statements/queries
FAIL	github.com/cockroachdb/cockroach/pkg/sql/logictest	10935.032s
=== RUN   TestSqlLiteLogic/fakedist/select4.test
=== PAUSE TestSqlLiteLogic/fakedist/select4.test
=== CONT  TestSqlLiteLogic/fakedist/select4.test

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2426222&tab=buildLog) on [master@be84ae4a59e6a5e50694c1756e05b0ba7c3a54d8](https://github.com/cockroachdb/cockroach/commits/be84ae4a59e6a5e50694c1756e05b0ba7c3a54d8):

Fatal error:
```
panic: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes [recovered]
	panic: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes

```

Stack:
```
goroutine 263239249 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc327a5de60, 0x5339860, 0xc01b1e0ac0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:206 +0x11f
panic(0x3d62a40, 0xc9af9b5e60)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5339920, 0xc7c41e72f0, 0xc1c1500ab0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:132 +0x1653
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1(0x5339920, 0xc7c41e72f0, 0xc1c1500ab0, 0x52a6280, 0xc7e1f9e280)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:707 +0x43
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc1c1500ab0, 0x5339920, 0xc7c41e72f0, 0xc2f7f3d620, 0xc1c1500ab0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:811 +0xd9
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0xc7852b80e0, 0x5339920, 0xc7c41e72f0, 0xc1e1ef7708, 0xc0000d6700, 0x14e6584c9b28)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:706 +0xfa
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc7b62ac780, 0x5339920, 0xc7c41e72f0, 0xc1e1ef7708, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:693 +0x10c
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc7b62ac780, 0x5339920, 0xc7c41e72f0, 0xc1e1ef7770, 0x4017be0, 0xc47089dd60)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:108 +0xbb
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).started(0xc7b62ac780, 0x5339920, 0xc7c41e72f0, 0x3, 0x3bf1e00)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:224 +0x62
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc6d45c97c0, 0x5339920, 0xc7c41e72f0, 0x3fb35e0, 0xc875e53a30, 0x52e9020, 0xc875e52dd8, 0xc7de85f020, 0xc7b62ac780, 0xc93a49b088, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1080 +0xf13
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc6d45c97c0, 0x5339860, 0xc59970c600, 0x52e9020, 0xc875e52dd8, 0xc7de85f020, 0xc709af6120, 0xc7b62ac780, 0xc93a49b088, 0x7, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:245 +0x3ce
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1(0x5339860, 0xc01b1e0ac0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:182 +0x1a9
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc327a5de60, 0x5339860, 0xc01b1e0ac0, 0xc0cf580a00, 0x1e, 0x0, 0xc709af62a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:346 +0x145
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:341 +0x11b

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestSqlLiteLogic/fakedist-spec-planning/slt_good_2.test#13
=== PAUSE TestSqlLiteLogic/fakedist-spec-planning/slt_good_2.test#13
=== CONT  TestSqlLiteLogic/fakedist-spec-planning/slt_good_2.test#13
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 8780 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 5219 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2433334&tab=buildLog) on [master@df21e5065a79d1dd5b0fd76018724dd26660beb2](https://github.com/cockroachdb/cockroach/commits/df21e5065a79d1dd5b0fd76018724dd26660beb2):

Fatal error:
```
panic: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes [recovered]
	panic: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes

```

Stack:
```
goroutine 35453346 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc063b586c0, 0x53460e0, 0xc00b1a8fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:206 +0x11f
panic(0x3d6d240, 0xca9c8fae40)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x53461a0, 0xc015873140, 0xc9c3efb170, 0x203277, 0x70)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:132 +0x1653
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc1a9e8e500, 0x53461a0, 0xc015873140, 0xc00f35b970, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:691 +0x9b
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc1a9e8e500, 0x53461a0, 0xc015873140, 0xc00f35b9d8, 0x4184ce0, 0xc9de73e460)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:108 +0xbb
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).SetDetails(0xc1a9e8e500, 0x53461a0, 0xc015873140, 0x4184ce0, 0xc9de73e460, 0x4071380, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:619 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/rowexec.SetResumeSpansInJob(0x53461a0, 0xc015873140, 0xca8f1526f0, 0x0, 0x1, 0x0, 0xc9c3efb170, 0xc1a9e8e500, 0x52545c, 0xc129d93470)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/backfiller.go:318 +0x1d4
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).distBackfill.func6(0x53461a0, 0xc015873140, 0xc9c3efb170, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:1051 +0x7a
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1(0x53461a0, 0xc015873140, 0xc9c3efb170, 0x3ff0000000000000, 0x54bd1c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:707 +0x43
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc9c3efb170, 0x53461a0, 0xc015873140, 0xc00f35bf88, 0xc9c3efb170, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:811 +0xd9
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0xc1930a1d50, 0x53461a0, 0xc015873140, 0xc00f35c078, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:706 +0xfa
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).distBackfill(0xc0908bfc70, 0x53461a0, 0xc015873140, 0x9, 0x2, 0xc350, 0x48d5150, 0xc12ffb32f0, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:1050 +0x4f2
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).backfillIndexes(0xc0908bfc70, 0x53461a0, 0xc015873140, 0x9, 0xc12ffb32f0, 0x1, 0x1, 0x1, 0xc0dbbffa78)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:1495 +0x193
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).runBackfill(0xc0908bfc70, 0x53461a0, 0xc015873140, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:300 +0x717
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).runStateMachineAndBackfill(0xc0908bfc70, 0x53461a0, 0xc015873140, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1446 +0x89
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).exec(0xc0908bfc70, 0x53461a0, 0xc129d934a0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:676 +0x523
github.com/cockroachdb/cockroach/pkg/sql.schemaChangeResumer.Resume.func1(0x3000000d9, 0xc000000000, 0xc0908bfc00, 0x53a7700)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:2100 +0x612
github.com/cockroachdb/cockroach/pkg/sql.schemaChangeResumer.Resume(0xc1a9e8e500, 0x53461a0, 0xc129d934a0, 0x3fbe200, 0xc1dc1fcbc8, 0xc147cf19e0, 0x0, 0xc061a324c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:2218 +0x7e3
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func1(0xc01bd3e5c0, 0x52f56a0, 0xc1dc1fcbc0, 0x53461a0, 0xc129d934a0, 0x3fbe200, 0xc1dc1fcbc8, 0xc147cf19e0, 0xc00f35d8f8)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1088 +0xde
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc150942dc0, 0x53461a0, 0xc129d93470, 0x3fbe200, 0xc1dc1fcbc8, 0x52f56a0, 0xc1dc1fcbc0, 0xc147cf19e0, 0xc1a9e8e500, 0xc0b4a301a0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1089 +0x87a
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc150942dc0, 0x53460e0, 0xc0628b0b80, 0x52f56a0, 0xc1dc1fcbc0, 0xc147cf19e0, 0xc1af4e6840, 0xc1a9e8e500, 0xc0b4a301a0, 0x7, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:245 +0x3ce
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1(0x53460e0, 0xc00b1a8fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:182 +0x1a9
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc063b586c0, 0x53460e0, 0xc00b1a8fc0, 0xc155eb0a00, 0x1e, 0x0, 0xc1af4e68a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:346 +0x145
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:341 +0x11b

```

<details><summary>Log preceding fatal error</summary><p>

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 4901 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 2900 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3262 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 4991 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 2970 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3515 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 120 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 120 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 120 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 121 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3319 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3323 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 239 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 173 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 5124 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 223 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 2983 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 215 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3519 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3399 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 312 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 347 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 5192 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 324 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 320 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 3007 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3522 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3423 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 357 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 408 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 367 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 374 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 5228 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 3019 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 3524 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3449 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 441 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 377 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 415 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 393 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 5256 statements/queries
*
* WARNING: disk slowness detected: unable to sync log files within 10s
*
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 406 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 430 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 5263 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_0.test: 3459 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 407 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 378 statements/queries

```

</p></details>

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2506238&tab=buildLog) on [master@258ef5765d0205f487b1481a9c26059db2a70362](https://github.com/cockroachdb/cockroach/commits/258ef5765d0205f487b1481a9c26059db2a70362):

```
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_21.test:20016: SELECT pk FROM tab2 WHERE col0 >= 649 AND (((col0 BETWEEN 361 AND 486) OR ((col0 IS NULL)) AND col1 IN (38.17,726.6,757.46) OR col1 >= 29.5 AND (col0 < 669) OR col0 < 530 OR col4 = 639.71 OR col4 = 119.43 AND col3 < 556 AND col0 > 736 OR col0 > 666 OR col0 >= 251 OR col1 >= 893.59 AND (((((col0 = 588))) AND col3 > 614 OR (col3 IN (SELECT col0 FROM tab2 WHERE col1 IN (75.42))) AND (col0 >= 884) AND col4 < 647.37) AND (col3 >= 78)))) ORDER BY 1 DESC
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(37); ()-->(12), (37)-->(11,37-39,41,42), (38,39)-->(37,41,42), (41)-->(37-39,42), (11)==(41), (41)==(11) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18387: MemoizeLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/statistics_builder.go:2649: rowsProcessed()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/memo.go:375: RowsProcessed()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:770: computeIndexLookupJoinCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:736: computeLookupJoinCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:456: ComputeCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:518: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201209 10:38:47.644753 153478109 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(37); ()-->(12), (37)-->(11,37-39,41,42), (38,39)-->(37,41,42), (41)-->(37-39,42), (11)==(41), (41)==(11) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(37); ()-->(12), (37)-->(11,37-39,41,42), (38,39)-->(37,41,42), (41)-->(37-39,42), (11)==(41), (41)==(11) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_21.test with config fakedist-disk: 3456 tests, 2 failures
E201209 10:38:47.644880 153478109 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_21.test:20023: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_21.test:20023: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_21.test:20023: too many errors encountered, skipping the rest of the input
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_21.test#01 (28.08s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2509095&tab=buildLog) on [master@c41a2ed503cd31d95bc3ee4365663811772c8bd3](https://github.com/cockroachdb/cockroach/commits/c41a2ed503cd31d95bc3ee4365663811772c8bd3):

```
=== CONT  TestSqlLiteLogic/fakedist-metadata/slt_good_22.test
    logic.go:2421: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test:23788: SELECT pk FROM tab2 WHERE col1 IN (SELECT col4 FROM tab2 WHERE ((col1 < 70.70 AND col3 <= 86 AND (col1 < 62.13) AND (((col3 BETWEEN 75 AND 76)) OR ((col1 = 49.52)) OR col3 > 39 OR (col4 IN (54.34,36.55,84.93,77.28,8.42,11.84))) AND (col3 > 77))))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(12,14), (12)-->(14) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18113: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1204: buildWhere()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1077: buildSelectClause()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1022: buildSelectStmtWithoutParens()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:995: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/with.go:29: processWiths()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:994: buildSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:265: buildStmt()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:229: buildStmtAtRoot()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:200: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:509: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:195: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:878: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:764: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:622: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:621: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201210 10:12:24.338284 113278661 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(12,14), (12)-->(14) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(12,14), (12)-->(14) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test with config fakedist-metadata: 4588 tests, 2 failures
E201210 10:12:24.338529 113278661 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test:23792: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test:23792: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test:23792: too many errors encountered, skipping the rest of the input
E201210 10:12:24.344533 148839002 sql/stats/automatic_stats.go:300  failed to refresh stats: node unavailable; try another peer
        --- FAIL: TestSqlLiteLogic/fakedist-metadata/slt_good_22.test (50.78s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2511935&tab=buildLog) on [master@5d9026c85bb9b762d7d02e2710ec36c31c641a81](https://github.com/cockroachdb/cockroach/commits/5d9026c85bb9b762d7d02e2710ec36c31c641a81):

```
        /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20469: SELECT pk FROM tab2 WHERE ((col3 IN (218,474,122))) OR ((((col1 = 515.97) AND (col0 <= 234) AND col3 > 773))) AND (col0 > 648) OR ((col0 IN (SELECT col3 FROM tab2 WHERE (((col4 <= 157.12) AND col0 = 988)) AND col1 <= 285.29) OR col0 IS NULL AND (((col0 > 776 OR (((((col1 > 399.78)))))))))) AND col3 >= 931 AND col4 <= 579.30 OR (col1 > 111.23 OR (((col3 IN (646,59,924,407)))))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18113: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:307: exploreSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:20: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:178: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:463: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201211 09:14:04.018925 110992547 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test with config fakedist-metadata: 4299 tests, 2 failures
E201211 09:14:04.019033 110992547 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: too many errors encountered, skipping the rest of the input
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_3.test: 5182 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-metadata/slt_good_1.test#09 (69.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2515106&tab=buildLog) on [master@960b4cfc54c8b78df56e62f07d7a07b986ceacff](https://github.com/cockroachdb/cockroach/commits/960b4cfc54c8b78df56e62f07d7a07b986ceacff):

```
NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2421: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18069: SELECT pk FROM tab2 WHERE col0 > 131 AND ((col0 <= 536)) OR col3 <= 184 AND (((col1 <= 851.25 OR col3 < 846 AND col3 > 64))) AND (col0 >= 423) AND col3 > 302 AND col4 <= 433.62 AND col0 IN (SELECT col3 FROM tab2 WHERE col0 > 550 AND (col4 < 281.61 AND col3 > 357))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18113: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:413: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:311: exploreSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:20: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:178: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:463: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:224: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:513: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:195: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:898: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:784: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:631: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:621: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201212 08:41:06.756805 68099772 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test with config fakedist-vec-off: 3687 tests, 2 failures
E201212 08:41:06.757083 68099772 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: too many errors encountered, skipping the rest of the input
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_9.test: 9055 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-vec-off/slt_good_8.test#01 (50.22s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2516741&tab=buildLog) on [master@53b08837cf5e76504e437bad80c97e05989d2c60](https://github.com/cockroachdb/cockroach/commits/53b08837cf5e76504e437bad80c97e05989d2c60):

```
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test:14756: SELECT pk FROM tab3 WHERE col3 < 846 AND (col3 > 65) OR col1 >= 656.53 AND col4 >= 317.41 OR (((col0 > 781 OR (col0 < 151 OR col1 < 930.12 OR (col1 IN (332.64,562.2) AND (col4 <= 937.4 AND (col4 < 217.83))) OR col4 IN (143.48,883.44,593.82,394.14) AND col3 < 677 AND col3 > 769 AND col3 < 894 OR col4 = 891.74 OR col3 < 710 AND col0 > 854 AND ((col4 = 727.43)) AND col3 = 18) OR (col3 > 398 OR (col0 BETWEEN 607 AND 828)) OR (col0 >= 173 AND (col0 > 623))))) AND ((((col4 IN (76.51,420.40,536.56,28.87,736.96,780.83)) OR (col0 > 487) AND ((col0 <= 166) AND col3 > 642 OR col0 <= 876 AND (((col0 = 100))) AND col3 > 576) OR col1 = 346.60)) AND (((col3 > 222 AND col3 >= 116 AND col0 >= 74))) OR (col0 IN (SELECT col3 FROM tab3 WHERE (col4 = 450.67)))) AND (col0 < 324) AND ((col3 IN (SELECT col0 FROM tab3 WHERE col0 >= 613 AND (col4 > 449.79) AND col3 < 868 AND col3 BETWEEN 474 AND 505 OR (col1 = 540.87))) AND col3 > 589 AND (col0 < 5) AND (col4 > 826.31)) OR col0 IS NULL AND col3 <= 178
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(73); ()-->(15), (73)-->(14,73-75,77,78), (77,78)-->(73-75), (74)-->(73,75,77,78), (14)==(74), (74)==(14) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18387: MemoizeLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/statistics_builder.go:2645: rowsProcessed()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/memo.go:375: RowsProcessed()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:770: computeIndexLookupJoinCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:736: computeLookupJoinCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/coster.go:456: ComputeCost()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:518: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201214 06:36:58.143896 78 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(73); ()-->(15), (73)-->(14,73-75,77,78), (77,78)-->(73-75), (74)-->(73,75,77,78), (14)==(74), (74)==(14) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(73); ()-->(15), (73)-->(14,73-75,77,78), (77,78)-->(73-75), (74)-->(73,75,77,78), (14)==(74), (74)==(14) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test with config local: 3068 tests, 2 failures
E201214 06:36:58.144003 78 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test:14760: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test:14760: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test:14760: too many errors encountered, skipping the rest of the input
        --- FAIL: TestSqlLiteLogic/local/slt_good_2.test#01 (35.40s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2519757&tab=buildLog) on [master@3018f66208f68fb90c77e3fad01f395a8f10ca8b](https://github.com/cockroachdb/cockroach/commits/3018f66208f68fb90c77e3fad01f395a8f10ca8b):

```
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_8.test#01
    logic.go:2421: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18069: SELECT pk FROM tab2 WHERE col0 > 131 AND ((col0 <= 536)) OR col3 <= 184 AND (((col1 <= 851.25 OR col3 < 846 AND col3 > 64))) AND (col0 >= 423) AND col3 > 302 AND col4 <= 433.62 AND col0 IN (SELECT col3 FROM tab2 WHERE col0 > 550 AND (col4 < 281.61 AND col3 > 357))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:38: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18108: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:413: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:311: exploreSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:20: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:178: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:463: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:224: Optimize()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:511: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:193: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:908: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:794: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:621: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201215 10:36:46.893745 153779716 sql/logictest/logic.go:3377  
pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
    logic.go:2166: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(14), (15)-->(11,15), (11)-->(15) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test with config fakedist-disk: 3687 tests, 2 failures
E201215 10:36:46.894013 153779716 sql/logictest/logic.go:3402  /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: too many errors encountered, skipping the rest of the input
    logic.go:2936: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: error while processing
    logic.go:2936: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test:18073: too many errors encountered, skipping the rest of the input
E201215 10:36:46.896503 177972669 sql/stats/automatic_stats.go:300  failed to refresh stats: node unavailable; try another peer
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_8.test#01 (45.67s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2522944&tab=buildLog) on [master@de91557c0634ab3797356d445ba39e37d45d8205](https://github.com/cockroachdb/cockroach/commits/de91557c0634ab3797356d445ba39e37d45d8205):

```
NOTE: internal errors may have more details in logs. Use -show-logs.
=== CONT  TestSqlLiteLogic/fakedist-vec-off/slt_good_2.test#02
    logic.go:2411: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_2.test:53604: SELECT pk FROM tab4 WHERE col1 IN (SELECT col4 FROM tab4 WHERE ((col0 > 50) AND col4 >= 85.31 AND col3 < 21))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(11,14), (11)-->(14) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:34: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18114: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1211: buildWhere()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1084: buildSelectClause()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1029: buildSelectStmtWithoutParens()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1002: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/with.go:29: processWiths()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:1001: buildSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:265: buildStmt()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:229: buildStmtAtRoot()
        github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:200: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:507: buildExecMemo()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:193: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:908: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:794: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:621: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201216 08:46:18.481345 67631054 sql/logictest/logic.go:3343  
pq: internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(11,14), (11)-->(14) (1)
    logic.go:2156: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(14); ()-->(15), (14)-->(11,14), (11)-->(14) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_2.test with config fakedist-vec-off: 8212 tests, 2 failures
E201216 08:46:18.481450 67631054 sql/logictest/logic.go:3368  /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_2.test:53608: too many errors encountered, skipping the rest of the input
    logic.go:2926: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_2.test:53608: error while processing
    logic.go:2926: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_2.test:53608: too many errors encountered, skipping the rest of the input
        --- FAIL: TestSqlLiteLogic/fakedist-vec-off/slt_good_2.test#02 (56.40s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2525468&tab=buildLog) on [master@eda9189cecbbc279f1857f6e6b992bdfd363397e](https://github.com/cockroachdb/cockroach/commits/eda9189cecbbc279f1857f6e6b992bdfd363397e):

```
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20469: SELECT pk FROM tab2 WHERE ((col3 IN (218,474,122))) OR ((((col1 = 515.97) AND (col0 <= 234) AND col3 > 773))) AND (col0 > 648) OR ((col0 IN (SELECT col3 FROM tab2 WHERE (((col4 <= 157.12) AND col0 = 988)) AND col1 <= 285.29) OR col0 IS NULL AND (((col0 > 776 OR (((((col1 > 399.78)))))))))) AND col3 >= 931 AND col4 <= 579.30 OR (col1 > 111.23 OR (((col3 IN (646,59,924,407)))))
        expected success, but found
        (XX000) internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
        func_dep.go:1513: in Verify()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/func_dep.go:1513: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/props/verify.go:50: Verify()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/check_expr.go:34: CheckExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/memo/expr.og.go:18114: MemoizeSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1078: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:589: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:1005: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:441: ConstructSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:307: exploreSelect()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.og.go:20: exploreGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/explorer.go:178: exploreGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:463: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:450: optimizeGroup()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:250: optimizeExpr()
        github.com/cockroachdb/cockroach/pkg/sql/opt/xform/optimizer.go:505: optimizeGroupMember()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E201217 07:12:30.742569 20129706 sql/logictest/logic.go:3343  
pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
    logic.go:2156: 
         pq: internal error: expected FD determinant and dependants to be disjoint: key(15); ()-->(11,14), (15)-->(12,15), (12)-->(15) (1)
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test with config local-spec-planning: 4299 tests, 2 failures
E201217 07:12:30.742696 20129706 sql/logictest/logic.go:3368  /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: too many errors encountered, skipping the rest of the input
    logic.go:2926: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: error while processing
    logic.go:2926: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_1.test:20473: too many errors encountered, skipping the rest of the input
        --- FAIL: TestSqlLiteLogic/local-spec-planning/slt_good_1.test#09 (54.81s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/opt/props/func_dep.go
+++ b/pkg/sql/opt/props/func_dep.go
@@ -959,6 +959,8 @@ func (f *FuncDepSet) ProjectCols(cols opt.ColSet) {
 			// Equivalence dependencies already maintain closure, so skip them.
 			if !fd.equiv {
 				fd.to = f.ComputeClosure(fd.to)
+				// Maintain the invariant that from and to columns don't overlap.
+				fd.to.DifferenceWith(fd.from)
 			}
 		}
 

--- a/pkg/sql/opt/props/func_dep_test.go
+++ b/pkg/sql/opt/props/func_dep_test.go
@@ -736,6 +736,18 @@ func TestFuncDeps_ProjectCols(t *testing.T) {
 	// Verify that lax keys convert to strong keys.
 	abcde.MakeNotNull(c(2, 3, 4, 5))
 	verifyFD(t, abcde, "key(3,4); (2)-->(3-5), (3,4)-->(2,5)")
+
+	// Regression test for #56358: ProjectCols was creating FD relations with
+	// overlapping from/to sets.
+	fd := &props.FuncDepSet{}
+	fd.AddConstants(c(2, 3))
+	fd.AddSynthesizedCol(c(4), 1)
+	fd.AddStrictKey(c(1), c(1, 2, 3, 4))
+	fd.AddEquivalency(2, 3)
+	verifyFD(t, fd, "key(1); ()-->(2,3), (4)-->(1), (1)-->(2-4), (2)==(3), (3)==(2)")
+	// Now project away column 3, and make sure we don't end up with (1)->(1,4).
+	fd.ProjectCols(c(1, 2, 4))
+	verifyFD(t, fd, "key(1); ()-->(2), (4)-->(1), (1)-->(4)")
 }
 
 func TestFuncDeps_AddFrom(t *testing.T) {

